--- a/target/linux/rockchip/image/armv8.mk
+++ b/target/linux/rockchip/image/armv8.mk
@@ -264,6 +264,17 @@ define Device/friendlyarm_nanopi-r6s
 endef
 TARGET_DEVICES += friendlyarm_nanopi-r76s
 
+define Device/friendlyarm_nanopi-zero2
+  DEVICE_VENDOR := FriendlyARM
+  DEVICE_MODEL := NanoPi Zero2
+  SOC := rk3328
+  UBOOT_DEVICE_NAME := nanopi-r2s-rk3328
+  IMAGE/sysupgrade.img.gz := boot-common | boot-script | pine64-bin | gzip | append-metadata
+  DEVICE_PACKAGES := -luci-app-gpsysupgrade kmod-gpio-button-hotplug kmod-usb-net-rtl8152
+  DEVICE_DTS = rockchip/rk3328-nanopi-r2s
+endef
+TARGET_DEVICES += friendlyarm_nanopi-zero2
+
 define Device/firefly_station-p2
   DEVICE_VENDOR := Firefly
   DEVICE_MODEL := Station P2

--- a/target/linux/rockchip/armv8/base-files/etc/board.d/02_network
+++ b/target/linux/rockchip/armv8/base-files/etc/board.d/02_network
@@ -67,6 +67,17 @@ rockchip_setup_interfaces()
 	friendlyarm,nanopi-r5s)
 		ucidef_set_interfaces_lan_wan 'eth1 eth2' 'eth0'
 		;;
+	friendlyarm,nanopi-neo3|\
+	friendlyarm,nanopc-t4|\
+	friendlyarm,nanopi-m4|\
+	friendlyarm,nanopi-m6|\
+	friendlyarm,nanopi-zero2|\
+	friendlyarm,nanopi-neo3-plus|\
+	friendlyarm,nanopi-neo4|\
+	friendlyarm,cm3588|\
+	friendlyarm,som-rk3399)
+		ucidef_set_interfaces_lan_wan 'wlan0' 'eth0'
+		;;
 	lyt,t68m)
 		ucidef_set_network_device_path lan1 'platform/fe2a0000.ethernet'
 		ucidef_set_network_device_path lan2 'platform/fe010000.ethernet'

--- a/target/linux/rockchip/image/Makefile
+++ b/target/linux/rockchip/image/Makefile
@@ -79,4 +79,16 @@ endif
 
 include $(SUBTARGET).mk
 
+define Image/Build
+	if [[ "$(PROFILE_SANITIZED)" == "friendlyarm_nanopi-zero2" ]]; then \
+		export IMG_PREFIX="$(IMG_PREFIX)$(if $(PROFILE_SANITIZED),-$(PROFILE_SANITIZED))"; \
+		export BIN_DIR=$(BIN_DIR); \
+		export TOPDIR=$(TOPDIR); \
+		export MORE=$(MORE); \
+		export ROOTFS_PARTSIZE=$(shell echo $$(($(ROOTFS_PARTSIZE)/1024/1024))); \
+		cd /data/packit/friendlywrt24-rk3528; \
+		. ~/packit/packit_nanopi.sh; \
+	fi
+endef
+
 $(eval $(call BuildImage))
