new file mode 100644
index 0000000000000..759baae3aeb18
--- /dev/null
+++ b/target/linux/mediatek/dts/mt7986a-xiaomi-redmi-router-ax6000.dts
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+
+/dts-v1/;
+#include "mt7986a-xiaomi-redmi-router-ax6000.dtsi"
+
+/ {
+	model = "Xiaomi Redmi Router AX6000";
+	compatible = "xiaomi,redmi-router-ax6000", "mediatek,mt7986a";
+};
+
+&spi_nand_flash {
+	mediatek,nmbm;
+	mediatek,bmt-max-ratio = <1>;
+	mediatek,bmt-max-reserved-blocks = <64>;
+};
+
+&partitions {
+			partition@580000 {
+				label = "crash";
+				reg = <0x580000 0x40000>;
+			};
+
+			partition@5c0000 {
+				label = "crash_log";
+				reg = <0x5c0000 0x40000>;
+			};
+
+			partition@600000 {
+				label = "ubi";
+				reg = <0x600000 0x6e00000>;
+			};
+};

--- a/target/linux/mediatek/base-files/lib/preinit/05_set_preinit_iface
+++ b/target/linux/mediatek/base-files/lib/preinit/05_set_preinit_iface
@@ -30,9 +30,9 @@ set_preinit_iface() {
 		;;
 	xiaomi,mi-router-ax3000t|\
 	xiaomi,mi-router-ax3000t-ubootmod|\
-	xiaomi,mi-router-wr30u-stock|\
+	xiaomi,mi-router-wr30u|\
 	xiaomi,mi-router-wr30u-ubootmod|\
-	xiaomi,redmi-router-ax6000-stock|\
+	xiaomi,redmi-router-ax6000|\
 	xiaomi,redmi-router-ax6000-ubootmod)
 		ip link set eth0 up
 		ifname=lan4
