--- a/target/linux/mediatek/dts/mt7981b-abt-asr3000.dts
+++ b/target/linux/mediatek/dts/mt7981b-abt-asr3000.dts
@@ -155,6 +155,10 @@
 		spi-tx-bus-width = <4>;
 		spi-rx-bus-width = <4>;
 
+		mediatek,nmbm;
+		mediatek,bmt-max-ratio = <1>;
+		mediatek,bmt-max-reserved-blocks = <64>;
+
 		partitions {
 			compatible = "fixed-partitions";
 			#address-cells = <1>;
@@ -208,13 +212,12 @@
 			partition@380000 {
 				label = "fip";
 				reg = <0x380000 0x200000>;
-				read-only;
 			};
 
 			partition@580000 {
 				compatible = "linux,ubi";
 				label = "ubi";
-				reg = <0x580000 0x7a80000>;
+				reg = <0x580000 0x6e80000>;
 
 				volumes {
 					ubi_rootdisk: ubi-volume-fit {
