--- a/package/feeds/luci/luci-mod-network/htdocs/luci-static/resources/view/network/dns.js
+++ b/package/feeds/luci/luci-mod-network/htdocs/luci-static/resources/view/network/dns.js
@@ -243,6 +243,11 @@ return view.extend({
 		o.optional = true;
 		o.placeholder = '/router.local/router.lan/192.168.0.1';
 
+		o = s.taboption('general', form.Flag, 'dns_redirect',
+			_('DNS Redirect'),
+			_('Force redirect all local DNS queries to DNSMasq, a.k.a. DNS Hijacking.'));
+		o.optional = true;
+
 		o = s.taboption('general', form.Flag, 'allservers',
 			_('All servers'),
 			_('Query all available upstream resolvers.') + ' ' + _('First answer wins.'));
@@ -437,6 +442,9 @@ return view.extend({
 		so.rmempty = false;
 		so.datatype = 'ipaddr("nomask")';
 
+		so = ss.option(form.Value, 'comments', _('Comments'));
+		so.rmempty  = true;
+
 		const ipaddrs = {};
 
 		Object.keys(hosts).forEach(function(mac) {